.github/scripts/trymerge.py

```diff
@@ -959,13 +988,46 @@ def check_for_sev(org: str, project: str, force: bool) -> None:
                 )
     return
 
+def fetch_check_run_conclusions(repo: GitRepo, commit: str) -> Dict[str, Tuple[str, str]]:
+    [owner, name] = repo.gh_owner_and_name()
+    checks = fetch_json_dict(f'https://api.github.com/repos/{owner}/{name}/commits/{commit}/check-runs')
+    check_run_conclusions = {}
+    if len(checks) == 0:
+        raise MandatoryChecksMissingError("Refusing to merge as land check(s) are not yet run")
+    for check_run in checks['check_runs']:
+        check_run_conclusions[check_run['name']] = (None if check_run['conclusion'] is None else check_run['conclusion'].upper(),
+                                                   check_run['html_url'])
+    return check_run_conclusions
+
+def validate_land_time_checks(repo: GitRepo, commit: str) -> None:
+    checks = fetch_check_run_conclusions(repo, commit)
+    [pending_checks, failed_checks] = categorize_checks(checks, checks)
+
+    if len(failed_checks) > 0:
+        raise RuntimeError(f"Failed to merge; some land checks failed: {checks_to_str(failed_checks)}")
+    if len(pending_checks) > 0:
+        raise MandatoryChecksMissingError(f"Refusing to merge as land check(s) {checks_to_str(pending_checks)} are not yet run")
+
+def categorize_checks(check_runs: Dict[str, Tuple[str, str]],
+                      required_checks: Iterable[str]) -> Tuple[List[Tuple[str, Optional[str]]], List[Tuple[str, Optional[str]]]]:
+    pending_checks: List[Tuple[str, Optional[str]]] = []
+    failed_checks: List[Tuple[str, Optional[str]]] = []
+    for checkname in required_checks:
+        if checkname not in check_runs:
+            pending_checks.append((checkname, None))
```

Comment:
this mean the job is never gonna run, right?

is this intended behavior

Reply:
this could mean that the check just hasn't been scheduled yet so we don't know if it'll run or not so it is intentional